app/models/rule_group_rule.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# Join table with the has-many-belongs-to-many relations between
+# RuleGroup and Rule
+class RuleGroupRule < ApplicationRecord
+  # These keep track of the parent-child relationship between
+  # rules and rule_groups
+  belongs_to :rule, optional: true
+  belongs_to :rule_group, optional: true
+  # These keep track of required/conflicting rules and rule groups
+  belongs_to :including, polymorphic: true, optional: true
+  belongs_to :requiring, polymorphic: true, optional: true
+  belongs_to :conflicting, polymorphic: true, optional: true
+  # These are needed to do ::RuleGroupRule.joins(rule/rule_group: :benchmark)
+  belongs_to :included_rule, class_name: 'Rule', foreign_key: 'including_id',
+                             inverse_of: :included_rules, optional: true
+  belongs_to :included_rule_group, class_name: 'RuleGroup', foreign_key: 'including_id',
+                                   inverse_of: :included_rule_groups, optional: true
+  # Need to use including_id because polymorphic assocations don't support computing the class
+  validates :including_id, allow_blank: true, uniqueness: { scope: %i[requiring_id conflicting_id] }
```

Comment:
@skateman When I tried to use `:including` instead of `:including_id`, there was an error message that said that polymorphic associations don't support computing the class so I made a comment for that